docs/contributor/image_mode.md

```diff
@@ -0,0 +1,217 @@
+    firewall-offline-cmd --zone=public --add-port=80/tcp && \
+    firewall-offline-cmd --zone=public --add-port=443/tcp && \
+    firewall-offline-cmd --zone=public --add-port=30000-32767/tcp && \
+    firewall-offline-cmd --zone=public --add-port=30000-32767/udp
+
+# Recursively make the root filesystem subtree shared as required by OVN images
+CMD ["/bin/sh", "-c", "mount --make-rshared / ; exec /sbin/init"]
+```
+
+> **Important:**<br>
+> When building a container image, podman uses host subscription information and
+> repositories inside the container. With only `BaseOS` and `AppStream` system
+> repositories enabled by default, we need to enable the `rhocp` and `fast-datapath`
+> repositories for installing MicroShift and its dependencies. These repositories
+> must be accessible in the host subscription, but not necessarily enabled on the host.
+
+Run the following image build command to create a local `bootc` image.
+
+Note that a pull secret file is used for:
+* The podman `--authfile` argument to pull the base `rhel-bootc:9.4` image from the `registry.redhat.io` registry
+* The build `PULL_SECRET` argument to be saved as `/etc/crio/openshift-pull-secret` file on the image for pulling MicroShift container images
+
+```
+PULL_SECRET=~/.pull-secret.json
+IMAGE_NAME=microshift-4.16-bootc
+
+sudo podman build --authfile "${PULL_SECRET}" -t "${IMAGE_NAME}" \
+    --build-arg PULL_SECRET="$(cat "${PULL_SECRET}")" \
+    -f Containerfile
+```
+
+Verify that the local MicroShift 4.16 `bootc` image was created.
+
+```
+$ sudo podman images "${IMAGE_NAME}"
+REPOSITORY                       TAG         IMAGE ID      CREATED        SIZE
+localhost/microshift-4.16-bootc  latest      193425283c00  2 minutes ago  2.31 GB
+```
+
+### Run Container
+
+Run the following command to start the MicroShift `bootc` image in an interactive
+terminal session. Note that the host container storage path is shared with the
+image so that it is reused in subsequent runs.
+
+```
+sudo podman run --rm -it --privileged \
+    -v /var/lib/containers/storage:/var/lib/containers/storage \
+    --name "${IMAGE_NAME}" \
+    "${IMAGE_NAME}"
+```
+
+Log into the container host when prompted using the `redhat:redhat` credentials.
+Run the following commands to start the MicroShift service and verify that all
+the pods are up and running without errors.
+
+```
+sudo systemctl start microshift
+sudo oc get pods -A -w \
+    --kubeconfig /var/lib/microshift/resources/kubeadmin/kubeconfig
+```
+
+> Run the `sudo shutdown now` command to stop the container host.
+
+## Run MicroShift Bootc Virtual Machine
+
+### Publish Image
+
+Log into the `development virtual machine` with the `microshift` user credentials.
+
+Run the following commands to log into the remote registry and publish the image.
+
+```
+REGISTRY_URL=quay.io
+REGISTRY_IMG=myorg/mypath/"${IMAGE_NAME}"
+
+sudo podman login "${REGISTRY_URL}"
+sudo podman push localhost/"${IMAGE_NAME}" "${REGISTRY_URL}/${REGISTRY_IMG}"
```

Comment:
FYI (but you probably know this already): `podman build --tag "${REGISTRY_URL}/${REGISTRY_IMG}"` to avoid intermediate `localhost/` tag.

Reply:
Yes, I do this in 2 steps because I only introduce remote registry in the second section. 
Let me tweak it a bit.